src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java

```diff
@@ -68,14 +69,18 @@ public void run() {
 
                 repo.saveAndFlush(letter);
 
-                logger.debug("Marked letter {} as Uploaded", letter.getId());
-            } catch (Exception e) {
-                logger.error("Exception uploading letter {}", letter.getId(), e);
+                logger.debug("Marked letter {} as {}", letter.getId(), letter.getStatus());
+            } catch (FtpException exception) {
+                logger.error("Exception uploading letter {}", letter.getId());
+
+                throw exception;
```

Comment:
I suggest we don't rethrow the FtpException but instead log it and break out of the foreach loop; if we rethrow the exception then we will roll back the transaction that may have uploaded some letters successfully and set them to the uploaded state.